bitcoind-rpc/src/main/scala/org/bitcoins/rpc/client/common/BitcoindRpcClient.scala

```diff
@@ -112,6 +120,82 @@ class BitcoindRpcClient(val instance: BitcoindInstance)(implicit
 
   override def downloadBlocks(
       blockHashes: Vector[DoubleSha256Digest]): Future[Unit] = FutureUtil.unit
+
+  override def processHeaders(headers: Vector[BlockHeader]): Future[ChainApi] =
+    Future.successful(this)
+
+  override def processFilterHeaders(
+      filterHeaders: Vector[FilterHeader],
+      stopHash: DoubleSha256DigestBE): Future[ChainApi] =
+    Future.successful(this)
+
+  override def getHeader(
+      hash: DoubleSha256DigestBE): Future[Option[BlockHeaderDb]] =
+    getBlockHeader(hash).map(header => Some(header.blockHeaderDb))
+
+  override def getHeadersBetween(
+      from: BlockHeaderDb,
+      to: BlockHeaderDb): Future[Vector[BlockHeaderDb]] = {
+    val headerFs =
+      from.height.to(to.height).map(height => getHeaderAtHeight(height))
+    Future.sequence(headerFs).map(_.toVector)
+  }
+
+  private def getHeaderAtHeight(height: Int): Future[BlockHeaderDb] =
+    for {
+      hash <- getBlockHash(height)
+      header <- getBlockHeader(hash)
+    } yield header.blockHeaderDb
+
+  override def getHeadersAtHeight(height: Int): Future[Vector[BlockHeaderDb]] =
+    getHeaderAtHeight(height).map(header => Vector(header))
+
+  override def getBestBlockHeader(): Future[BlockHeaderDb] =
+    for {
+      hash <- getBestBlockHash
+      header <- getBlockHeader(hash)
+    } yield header.blockHeaderDb
+
+  override def nextBlockHeaderBatchRange(
+      prevStopHash: DoubleSha256DigestBE,
+      batchSize: Int): Future[Option[FilterSyncMarker]] = FutureUtil.none
+
+  override def nextFilterHeaderBatchRange(
+      stopHash: DoubleSha256DigestBE,
+      batchSize: Int): Future[Option[FilterSyncMarker]] = FutureUtil.none
+
+  override def processFilters(
+      message: Vector[CompactFilterMessage]): Future[ChainApi] =
+    Future.successful(this)
+
+  override def processCheckpoints(
+      checkpoints: Vector[DoubleSha256DigestBE],
+      blockHash: DoubleSha256DigestBE): Future[ChainApi] =
+    Future.successful(this)
+
+  override def getFilterHeaderCount(): Future[Int] = ???
```

Comment:
Since we just using bitcoind in this scenario, and bitcoind generates the filters`, it's probably safe to assume that `filterHeaderCount == blockHeaderCount`? So you might be able to just use `getBlockCount()` behind the scenes? I would put this reasoning as a scaladoc so we don't forget in the future.

Reply:
These are `???` because on versions without filter support we want to return an error instead.

Reply:
Context for how it was done last time: https://github.com/bitcoin-s/bitcoin-s/pull/2087

Reply:
Ohhh, didn't think about filter support only being post v18 i think? Can we do something with versions to actually support this in newer versions?

Reply:
Yeah, it's overridden in those